utils/ssh.go

```diff
@@ -0,0 +1,127 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+const (
+	// defaultSSHPort is the default port used for ssh.
+	defaultSSHPort = 22
+	// defaultSSHUser is the default user user for ssh.
+	defaultSSHUser = "core"
+	// defaultSSHTimeout is the default maximum amount of time for the ssh connection to establish.
+	defaultSSHTimeout = 30 * time.Second
+)
+
+type SSHConfig struct {
+	// default to `core`
+	User string
+	// defaults to `22`
+	Port int32
+	// uses `SSH_AUTH_SOCK` if this is empty
+	IdentifyKeyFile string
+	// Timeout is the maximum amount of time for the ssh connection to establish.
+	// defaults to `1 minute`
+	Timeout time.Duration
+}
+
+// SSHClient holds config for ssh.
+type SSHClient struct {
+	*ssh.ClientConfig
+	Port int32
+}
+
+// MustNewSSHClient returns *SSHClient.
+// Uses default values for SSHConfig.
+func MustNewSSHClient(config *SSHConfig) *SSHClient {
+	var authMethod ssh.AuthMethod
+	sock := os.Getenv("SSH_AUTH_SOCK")
```

Comment:
why we try to read this from env, not config? we need to document this if there is a reason.